src/memory.c

```diff
@@ -65,44 +65,27 @@
 static char const *const label_state = "system.memory.state";
 
 typedef enum {
-  COLLECTD_MEMORY_TYPE_USED,
-  COLLECTD_MEMORY_TYPE_FREE,
-  COLLECTD_MEMORY_TYPE_BUFFERS,
-  COLLECTD_MEMORY_TYPE_CACHED,
-  COLLECTD_MEMORY_TYPE_SLAB_TOTAL,
-  COLLECTD_MEMORY_TYPE_SLAB_RECL,
-  COLLECTD_MEMORY_TYPE_SLAB_UNRECL,
-  COLLECTD_MEMORY_TYPE_WIRED,
-  COLLECTD_MEMORY_TYPE_ACTIVE,
-  COLLECTD_MEMORY_TYPE_INACTIVE,
-  COLLECTD_MEMORY_TYPE_KERNEL,
-  COLLECTD_MEMORY_TYPE_LOCKED,
-  COLLECTD_MEMORY_TYPE_ARC,
-  COLLECTD_MEMORY_TYPE_UNUSED,
-  COLLECTD_MEMORY_TYPE_AVAILABLE,
-  COLLECTD_MEMORY_TYPE_USER_WIRE,
-  COLLECTD_MEMORY_TYPE_LAUNDRY,
-  COLLECTD_MEMORY_TYPE_MAX, /* #states */
+  STATE_USED,
+  STATE_FREE,
+  STATE_SHARED,
+  STATE_BUFFERS,
+  STATE_CACHED,
+  STATE_WIRED,
+  STATE_ACTIVE,
+  STATE_INACTIVE,
+  STATE_KERNEL,
+  STATE_LOCKED,
+  STATE_ARC,
+  STATE_UNUSED,
+  STATE_USER_WIRE,
+  STATE_LAUNDRY,
+  STATE_MAX, /* #states */
 } memory_type_t;
 
-static char const *memory_type_names[COLLECTD_MEMORY_TYPE_MAX] = {
-    "used",
-    "free",
-    "buffers",
-    "cached",
-    "slab",
-    "slab_reclaimable",
-    "slab_unreclaimable",
-    "wired",
-    "active",
-    "inactive",
-    "kernel",
-    "locked",
-    "arc",
-    "unusable",
-    "available",
-    "user_wire",
-    "laundry",
+static char const *memory_type_names[STATE_MAX] = {
+    "used",  "free",     "shared",    "buffers", "cached",
+    "wired", "active",   "inactive",  "kernel",  "locked",
+    "arc",   "unusable", "user_wire", "laundry",
```

Comment:
IMHO this would be easier to read / check against each other if they would be sorted alphabetically.

Reply:
Sound good; done.